packages/lubuild/README.md

```diff
@@ -0,0 +1,118 @@
+1. if there isn't a LUIS AppId for the model, it will create one
+2. it runs ludown tool to generate the model
+3. it uploads the model to the appid
+4. it trains the model
+5. it publishes it
+6. it then outputs a .dialog definition of a LuisRecognizer configured to use that model.
+
+## Installation
+
+```npm install -g lubuild```
+
+## Usage
+1. create a models.config with **name** and **models** configured to your root .LU files
+2. Get LUIS.ai authoringKey
+3. invoke 
+     
+    ```lubuild --authoringKey YOURKEY```
+
+**LUBuild** will create all of the assets you need from your local .LU files
+
+## LU and language varations files
+Every LU file can have multiple language variations.  **LUBuild** will build a model for each one.  
+
+The pattern for .LU files and the language variants are
+```json
+example.en-us.lu
+example.fr-fr.lu
+example.de-de.lu
+etc.
+````
+Each one of these .lu files will have a language specific model created fo rit.
+
+## LUIS Applications 
+
+Every combination of project, environment, and file name (With locale) will be used
+to name the application created on your behalf.
+
+LUIS application names will 
+> {projectname}-{environment}-{LUfilename}
+
+Example:
+```
+    MyProject(tomlm)-Contoso.dialog1.en-us.lu
+    MyProject(tomlm)-Contoso.dialog1.fr-fr.lu
+    MyProject(tomlm)-Contoso.dialog1.de-de.lu
+```
+The same application name will be used in each azure region, with endpoints internal to it.
+
+## Generated .Dialog file for each lu file
+All language variations of a .LU files with the same prefix will get a .dialog file created
+which is a LUIS recognizer configured for it. 
+
+Example:
+```
+    models/tomlm/westus/Contoso.dialog1.lu.dialog <-- MultiLanguageRecognizer configured to use all of the languages 
+    models/tomlm/westus/Contoso.dialog1.en-us.lu.dialog <-- LuisRecognizer 
+    models/tomlm/westus/Contoso.dialog1.fr-fr.lu.dialog <-- LuisRecognizer 
+    models/tomlm/westus/Contoso.dialog1.de-de.lu.dialog <-- LuisRecognizer 
+```
+
+The net reslut is that to consome all of the language models in a dialog you simply can do this:
+Example:
+```json
+{
+    "$type":"Microsoft.AdaptiveDialog",
+    "recognizer": "Contoso.dialog1.lu"  <-- this will be the multilanguage model with all variations
+}
+```
+
+This will configure your recognizer to a LURecognizer("Contsoso.Dialog1.lu") which internally
+will use your memory *settings.luis.projectname* and *settings.luis.environment* settings to
+bind to the correct .dialog file for your model and runtime environment.
+
+## Environments
+When multiple people are working with models you want to be able to work with models
+independently from each other tied to the source control.  **LUBuild**
+
+By default, **LUBuild** uses the logged in user alias as the enviornment, so that you can
+simply create and work with local changes.  
+
+The environment is placed in *settings.luis.environment* configuration to control which
```

Comment:
where is this file get saved / managed? Should we expect developers to edit this file?

Reply:
it gets saved as a peer file to appsettings.json